lib/compile/util.js

```diff
@@ -214,20 +214,33 @@ function getPath(currentPath, prop, jsonPointers) {
 }
 
 
+var JSON_POINTER = /^#|\/(?:[^~]|~0|~1)*?$/;
```

Comment:
Why do you need `"#"` in the pattern? If you mean it as the pointer to the root it should be `""` (an empty string) instead. `"/"` is the pointer to the empty property at the top level. `"#"` in relative json pointer is a special thing resolving to the name of the property/index rather than to data. Or am I missing something?